### PR TITLE
docs: refresh docs landing page

### PR DESCRIPTION
**Pull request by richardjgowers**
provides short description of the package on landing page

reworked landing page cards to match current contents better

Checklist
* [ ] Added a ``news`` entry

## Developers certificate of origin
- [ ] I certify that this contribution is covered by the MIT License [here](https://github.com/OpenFreeEnergy/openfe/blob/main/LICENSE) and the **Developer Certificate of Origin** at <https://developercertificate.org/>.

### CODE REVIEW

**Review by jameseastwood — APPROVED** [review]
I did some copy editing and it looks good to me now. I also checked that all the links work and go to the right place.
image

File: docs/index.rst [hunk 1]
```diff
@@ -6,81 +6,89 @@
...
-        :link: reference/api/openmm_rfe
+        :link: reference/index
         :link-type: doc
 
-        Documentation for OpenFE's OpenMM-based Hybrid Topology Relative Free
-        Energy Protocol.
+        Technical details on the toolkit's core methods and classes
 
 
 .. toctree::
```
Comment (RiesBen) [inline docs/index.rst:94]: stupid question do we need this here?
```suggestion

```

I think it makes this, right?
image

**Review by RiesBen — APPROVED** [review]
I like the improvements, I have one minor comment.

File: docs/index.rst [hunk 2]
```diff
@@ -6,81 +6,89 @@
 Welcome to the OpenFE documentation!
 ====================================
 
+The **OpenFE** toolkit provides a free and open-source framework for calculating alchemical free energies.
```
Comment (hannahbaumann) [inline docs/index.rst:9]: I don't think "alchemical free energies" is something that exists?? But I could be wrong!
Maybe
```suggestion
The **OpenFE** toolkit provides a free and open-source framework for alchemical free energy calculations.
```

File: docs/index.rst (hunk 1 above)
Comment (richardjgowers) [reply docs/index.rst:94]: seems to create warnings without 
```
checking consistency... /home/richard/code/openfe/docs/CHANGELOG.rst: WARNING: document isn't included in any toctree
/home/richard/code/openfe/docs/cookbook/index.rst: WARNING: document isn't included in any toctree
/home/richard/code/openfe/docs/guide/index.rst: WARNING: document isn't included in any toctree
/home/richard/code/openfe/docs/installation.rst: WARNING: document isn't included in any toctree
/home/richard/code/openfe/docs/reference/index.rst: WARNING: document isn't included in any toctree
/home/richard/code/openfe/docs/tutorials/index.rst: WARNING: document isn't included in any toctree
```

**Pull request merged**